.github/workflows/ingestion-service.yml

```
@@ -0,0 +1,101 @@
+name: Ingestion Service Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        required: false
+        description: a tag that will be used for identifying this version (xaynetci/xayn_discovery_ingestion_service:$TAG)
+
+permissions:
+  contents: read
+
+env:
+  DENY_WARNINGS: false
+
+jobs:
+  rust-build:
+    runs-on: hetzner-pm
+    container:
+      image: xaynetci/yellow:v4
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup job
+        uses: ./.github/actions/setup-job-docker
+        with:
+          rust: true
+
+      - name: Build artifacts
+        run: just build-ingestion-service
+
+      - name: Download assets
+        run: just download-assets
+
+      - name: Prepare temp directory
+        working-directory: ${{ runner.temp }}
+        run: |
+          rm -rf ./ingestion-service
+          mkdir -p ingestion-service/assets
+
+      - name: Create archive
+        working-directory: ${{ runner.temp }}
+        run: |
+          cp ${GITHUB_WORKSPACE}/${{ env.RUST_WORKSPACE }}/target/release/ingestion ./ingestion-service/ingestion
+          cp ${GITHUB_WORKSPACE}/${{ env.RUST_WORKSPACE }}/web-api/Dockerfile.ingestion ./ingestion-service/Dockerfile
+          cp ${GITHUB_WORKSPACE}/${{ env.FLUTTER_WORKSPACE }}/example/assets/smbert_v0001/smbert-quantized.onnx ./ingestion-service/assets/model.onnx
+          cp ${GITHUB_WORKSPACE}/${{ env.FLUTTER_WORKSPACE }}/example/assets/smbert_v0001/vocab.txt ./ingestion-service/assets/vocab.txt
+          tar -cvf ingestion-service.tar ./ingestion-service
+
+      - name: Upload archive
+        uses: actions/upload-artifact@v3
```

:wrench:  action should be pinned

```suggestion
        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
```